### PR TITLE
markdown-it emoji plugin

### LINKED ISSUE #3: markdown-it-emoji plugin
Hey @digitalmoksha 👋 

Awesome port 👏 I would like to ask if you are interested in a `markdown-it-emoji` plugin?

📦 https://github.com/markdown-it/markdown-it-emoji

I didn't checked all the implementation in detail, but i guess i could volunteer. 🤓

### PR DESCRIPTION
**Pull request by neumayr**
* markdown-it emoji plugin
_Ported from the [javascript version](https://github.com/markdown-it/markdown-it-emoji). Synced with v1.4.0._

🤓huge thanks to @digitalmoksha ✨awesome work 🚀 

Closes #3

### CODE REVIEW

**Comment by neumayr** [comment]
Awesome, first I was confused because locally some old test failed to run. 👏
But now I see they also fail on Travis 🙈

**Comment by digitalmoksha** [comment]
@neumayr yes, there are a couple tests in the other plugins that I've been unable to track down, but it's a couple edge cases.

I think the emoji plugin should be done, just updated the readme.  Take a look - if everything looks good i'll accept the pull request and merge it!

File: lib/motion-markdown-it-plugins/emoji/plugin/data/light.rb [hunk 1]
```diff
@@ -0,0 +1,161 @@
+module MotionMarkdownItPlugins
+  module EmojiPlugin
+    module Data
+      module Light
+        EMOJIES_DEF_LIGHT = {
+          "grinning" => "😀",
```
Comment (neumayr) [inline lib/motion-markdown-it-plugins/emoji/plugin/data/light.rb:6]: What do you think about using ruby symbols as hash keys? If you want avoid hashrocket syntax (`=>`)

File: lib/motion-markdown-it-plugins/emoji/plugin/data/light.rb [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+          "white_medium_square" => "◻️",
+          "black_large_square" => "⬛️",
+          "white_large_square" => "⬜️",
+          "black_joker" => "🃏",
+          "mahjong" => "🀄️"
+        }.freeze
+      end
+    end
+  end
+end
```
Comment (neumayr) [inline lib/motion-markdown-it-plugins/emoji/plugin/data/light.rb]: 💅 `missing newline at end of file`

**Comment by neumayr** [comment]
Awesome work, Thank you for this one 🙇 

I've just had some small bits, you can decide on.

File: lib/motion-markdown-it-plugins/emoji/plugin/data/light.rb (hunk 1 above)
Comment (digitalmoksha) [reply lib/motion-markdown-it-plugins/emoji/plugin/data/light.rb:6]: Because of these in `full.rb`

```ruby
          "100" => "💯",
          "1234" => "🔢",
```

we can't convert all of them into symbols.  However I did make a change to allow the options passed in to be symbols.

File: lib/motion-markdown-it-plugins/emoji/plugin/data/light.rb (hunk 2 above)
Comment (digitalmoksha) [reply lib/motion-markdown-it-plugins/emoji/plugin/data/light.rb]: Thanks, fixed ;-)

File: lib/motion-markdown-it-plugins/emoji/plugin/normalize_opts.rb [hunk 3]
```diff
@@ -6,9 +6,16 @@ def quoteRE(str)
         str.gsub(/([.?*+^$\[\]\\(){}|-])/, '\\\\\1')
       end
 
+      # convert keys from symbols into strings
+      #------------------------------------------------------------------------------
+      def stringify(symbol_hash)
+        symbol_hash.collect{|k,v| [k.to_s, v]}.to_h
+      end
```
Comment (neumayr) [inline lib/motion-markdown-it-plugins/emoji/plugin/normalize_opts.rb:13]: just fyi - Rails ships with a [stringify_keys](https://apidock.com/rails/Hash/stringify_keys) method. And the core function was added to [Ruby 2.5 adds Hash#transform_keys method](https://blog.bigbinary.com/2018/01/09/ruby-2-5-adds-hash-transform_keys-method.html) lately.

Comment (digitalmoksha) [reply lib/motion-markdown-it-plugins/emoji/plugin/normalize_opts.rb:13]: True, but since I'm supporting RubyMotion as well, I can't rely on Rails or Ruby 2.5. And I want to keep, for now, dependencies small.
